include/boost/unordered/detail/foa.hpp

```diff
@@ -39,12 +39,24 @@
 #include <type_traits>
 #include <utility>
 
-#if defined(__SSE2__)||\
+#if !defined(BOOST_UNORDERED_DISABLE_SSE2)&& \
+    !defined(BOOST_UNORDERED_ENABLE_SSE2)
+#if defined(__SSE2__)|| \
     defined(_M_X64)||(defined(_M_IX86_FP)&&_M_IX86_FP>=2)
-#define BOOST_UNORDERED_SSE2
+#define BOOST_UNORDERED_ENABLE_SSE2
```

Comment:
I don't like mixing up macros defined by the user with macros defined by the library. ENABLE/DISABLE should only be set by the user and never by the library. We should retain our own macro and define it accordingly.